### PR TITLE
fix(RHTAPBUGS-1019): add support for staged index use case

### PR DESCRIPTION
**Pull request by scoheb**
- set parameters accordingly when stagedIndex = true.
- we want to force opt_in to false and ensure no images are overwritten or published

Related to https://github.com/redhat-appstudio/release-service-catalog/pull/305

### CODE REVIEW

**Review by happybhati — APPROVED** [review]

File: internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml [hunk 1]
```diff
@@ -131,13 +135,19 @@ spec:
...
             mustPublishIndexImage="true"
         else
-            echo "Fragment has \`fbc_opt_in==${fbcOptIn}\`"
-            # in case of more than one image all should have the same fbc_opt_in value
-            if [ $(wc -w <<< ${fbcOptIn[@]}) -ne 1 ]; then
-                fbcOptIn="false"
-                mustSignIndexImage="false"
-                mustPublishIndexImage="false"
-            fi
+            if [ "$(params.stagedIndex)" == "true" ]; then
```
Comment (mmalina) [inline internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml:138]: Please give it a thought as well, but I think we can remove two levels of nesting like this:
```sh
if [ "$(params.hotfix)" == "true" ]
...
elif [ "$(params.stagedIndex)" == "true" ]
...
elif [ $(wc -w <<< ${fbcOptIn[@]}) -ne 1 ]
...
fi
```
And then perhaps add the echo after all of that:
```sh
echo "Fragment has \`fbc_opt_in==${fbcOptIn}\`"
```

**Comment by mmalina** [comment]
> Requires [redhat-appstudio/release-service-catalog#305](https://github.com/redhat-appstudio/release-service-catalog/pull/305)

Since the new param is optional and has a default, this doesn't really require the other one. It's rather the opposite - this one is needed by 305, right? It's a detail, but I was both of them saying they require the other which seemed odd.

File: internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml [hunk 2]
```diff
@@ -4,7 +4,7 @@ kind: Task
 metadata:
   name: t-add-fbc-fragment-to-index-image
   labels:
-    app.kubernetes.io/version: "0.1"
+    app.kubernetes.io/version: "0.1.1"
```
Comment (mmalina) [inline internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml:7]: It's a new functionality, so probably 0.2.0?

Comment (scoheb) [reply internal-services/catalog/iib-add-fbc-fragment-to-index-image-task.yaml]: fixed

**Review by theflockers — APPROVED** [review]

**Pull request merged**